src/renderer/components/wallet/assets/AssetDetails.tsx

```diff
@@ -95,8 +97,67 @@ export const AssetDetails: React.FC<Props> = (props): JSX.Element => {
     [loadTxsHandler]
   )
 
+  const oRuneAsset: O.Option<Asset> = useMemo(() => FP.pipe(oAsset, O.filter(isRuneBnbAsset)), [oAsset])
+
+  const isRuneAsset: boolean = useMemo(() => FP.pipe(oRuneAsset, O.isSome), [oRuneAsset])
+
+  const oRuneBalance: O.Option<WalletBalance> = useMemo(() => getWalletBalanceByAsset(oBalances, oRuneAsset), [
+    oRuneAsset,
+    oBalances
+  ])
+
+  const oRuneAmount: O.Option<BaseAmount> = useMemo(
+    () =>
+      FP.pipe(
+        oRuneBalance,
+        O.map(({ amount }) => amount)
+      ),
+    [oRuneBalance]
+  )
+
+  const oRuneAddress: O.Option<Address> = useMemo(
+    () =>
+      FP.pipe(
+        oRuneBalance,
+        O.map(({ walletAddress }) => walletAddress)
+      ),
+    [oRuneBalance]
+  )
+
+  const runeUpgradeDisabled: boolean = useMemo(
+    () =>
+      isRuneAsset &&
+      FP.pipe(
+        oRuneAmount,
+        O.map((amount) => amount.amount().isLessThanOrEqualTo(0)),
+        O.getOrElse<boolean>(() => true)
+      ),
+    [isRuneAsset, oRuneAmount]
+  )
+
+  const actionColSpanDesktop = isRuneAsset ? 8 : 12
+  const actionColSpanMobile = 24
+
+  const upgradeRune = useCallback(
+    (_) =>
+      FP.pipe(
+        sequenceTOption(oRuneAsset, oRuneAmount, oPoolAddress, oRuneAddress),
+        O.map(([asset, amount, recipient, runeAddress]) => {
+          console.log('upgradeRuneHandler', recipient, amount, asset, runeAddress)
+          upgradeRuneHandler({ recipient, amount, asset, memo: `SWITCH:${runeAddress}` })
```

Comment:
`SWITCH` memo will be part of `asgardex-util` (see [Add SWITCH memo !16](https://gitlab.com/thorchain/asgardex-common/asgardex-util/-/issues/16))